assets/src/edit-story/components/panels/border/borderWidth.js

```diff
@@ -0,0 +1,173 @@
+  font-size: ${({ theme }) => theme.fonts.body2.size};
+  line-height: ${({ theme }) => theme.fonts.body2.lineHeight};
+  letter-spacing: ${({ theme }) => theme.fonts.body2.letterSpacing};
+  text-align: center;
+  width: 100%;
+  display: inline-block;
+  margin-top: 8px;
+  cursor: pointer;
+`;
+
+const ToggleWrapper = styled.div`
+  height: 60px;
+`;
+
+function WidthControls({ selectedElements, pushUpdateForObject }) {
+  const border = useCommonObjectValue(
+    selectedElements,
+    'border',
+    DEFAULT_BORDER
+  );
+
+  // Only if true for all selected elements.
+  const lockBorder = border.lockedWidth === true;
+
+  const handleChange = useCallback(
+    (name, value) => {
+      const newBorder = !lockBorder
+        ? {
+            [name]: value,
+          }
+        : {
+            left: value,
+            top: value,
+            right: value,
+            bottom: value,
+          };
+      pushUpdateForObject('border', newBorder, DEFAULT_BORDER, true);
+    },
+    [pushUpdateForObject, lockBorder]
+  );
+
+  const handleLockChange = useCallback(
+    (newBorder) => {
+      pushUpdateForObject('border', newBorder, DEFAULT_BORDER, true);
+    },
+    [pushUpdateForObject]
+  );
+
+  return (
+    <Row>
+      <Label>
+        <BoxedNumeric
+          value={border.left}
+          onChange={(value) => handleChange('left', value)}
+          aria-label={__('Edit: Left border', 'web-stories')}
+        />
+        <LabelText>{__('Left', 'web-stories')}</LabelText>
+      </Label>
+      <Space />
+      <Label>
+        <BoxedNumeric
+          value={border.top}
+          onChange={(value) => handleChange('top', value)}
+          aria-label={__('Edit: Top border', 'web-stories')}
+        />
+        <LabelText>{__('Top', 'web-stories')}</LabelText>
+      </Label>
+      <Space />
+      <Label>
+        <BoxedNumeric
+          value={border.right}
+          onChange={(value) => handleChange('right', value)}
+          aria-label={__('Edit: Right border', 'web-stories')}
+        />
+        <LabelText>{__('Right', 'web-stories')}</LabelText>
+      </Label>
+      <Space />
+      <Label>
+        <BoxedNumeric
+          value={border.bottom}
+          onChange={(value) => handleChange('bottom', value)}
+          aria-label={__('Edit: Bottom border', 'web-stories')}
+        />
+        <LabelText>{__('Bottom', 'web-stories')}</LabelText>
+      </Label>
+      <Space />
+      <ToggleWrapper>
+        <Toggle
+          icon={<Lock />}
+          uncheckedIcon={<Unlock />}
+          value={lockBorder}
+          onChange={() => {
+            let args = {
+              lockedWidth: !lockBorder,
+            };
+            // If the border width wasn't locked before (and is now), unify all the values.
+            if (!lockBorder) {
+              args = {
+                ...args,
+                top: border.left,
+                right: border.left,
+                bottom: border.left,
+              };
```

Comment:
Sam confirmed that the lock behavior should be similar to padding lock behavior (except for hiding all the inputs) -- since the first input takes precedence there then I'm assuming it should take precedence here, too. Thoughts?

Reply:
Just ask Sam (and/or have it added to the Product Brief) to cover behinds 🙈

Reply:
Well, I thought I did already, will confirm again.

Reply:
Yes, confirmed that we'll use the first input's value.